test/e2e/scheduler_predicates.go

```diff
@@ -900,16 +886,37 @@ func getRequestedCPU(pod v1.Pod) int64 {
 	return result
 }
 
-func waitForScheduler() {
-	// Wait a bit to allow scheduler to do its thing
-	// TODO: this is brittle; there's no guarantee the scheduler will have run in 10 seconds.
-	framework.Logf("Sleeping 10 seconds and crossing our fingers that scheduler will run in that time.")
-	time.Sleep(10 * time.Second)
+func waitForScheduler(f *framework.Framework, PodName string, expectSuccess bool) {
+	framework.Logf("Waiting for scheduler...")
+	action := func() error { // No action before waiting for Scheduler.
+		return nil
```

Comment:
The action here should be to create the pod you expect to observe. Otherwise there is a race between starting the watch and scheduler binding.

The `observeEventAfterAction` function is supposed to take care of the rest of the ordering concerns (make sure the watch is alive before evaluating the `action` thunk.)

Side note: the sibling helper function for observing node updates has additional synchronization barriers to try to guarantee that interleaving. It's probably a good idea to bring them to parity by adding the informer started guard that is signaled from the first invocation of the underlying informer's watch.